GANDLF/utils/gandlf_logger.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import yaml
+from pathlib import Path
+from importlib import resources
+import colorlog
+import tempfile
+from GANDLF.utils import get_unique_timestamp
+
+
+def _flush_to_console():
+    formatter = colorlog.ColoredFormatter(
+        "%(log_color)s%(asctime)s - %(levelname)s - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+        log_colors={
+            "DEBUG": "blue",
+            "INFO": "green",
+            "WARNING": "yellow",
+            "ERROR": "red",
+            "CRITICAL": "bold_red",
+        },
+    )
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatter)
+    logging.root.setLevel(logging.DEBUG)
+    logging.root.addHandler(console_handler)
+
+
+def _create_tmp_log_file():
+    tmp_dir = Path(tempfile.gettempdir())
+    log_dir = Path.joinpath(tmp_dir, ".gandlf")
+    log_dir.mkdir(parents=True, exist_ok=True)
+    log_file = Path.joinpath(log_dir, get_unique_timestamp() + ".log")
+    _create_log_file(log_file)
+    return log_file
+
+
+def _create_log_file(log_file):
+    log_file = Path(log_file)
+    log_file.write_text("Starting GaNDLF logging session \n")
+
+
+def _save_logs_in_file(log_file, config_path):
+    with resources.open_text("GANDLF", config_path) as file:
+        config_dict = yaml.safe_load(file)
+        config_dict["handlers"]["rotatingFileHandler"]["filename"] = str(log_file)
+        logging.config.dictConfig(config_dict)
+
+
+def gandlf_logger_setup(log_file=None, config_path="logging_config.yaml"):
```

Comment:
Since this is an externally visible function, I would suggest adding the return to helping typing suggestions/hints.

```suggestion
def gandlf_logger_setup(log_file=None, config_path="logging_config.yaml") -> None:
```

Reply:
Yes, you are right